credentials/endpointcreds/internal/client/middleware.go

```diff
@@ -86,6 +89,7 @@ func (d *deserializeOpGetCredential) HandleDeserialize(ctx context.Context, in s
 	if !ok {
 		return out, metadata, &smithy.DeserializationError{Err: fmt.Errorf("unknown transport type %T", out.RawResponse)}
 	}
+	defer response.Body.Close()
```

Comment:
Can you verify this, probably don't need to close the body here since you have AddCloseResponseBodyMiddleware to close the response body.

Reply:
Ah thanks for catching this I thought I had removed these.